libraries/ortbConverter/converter.js

```diff
@@ -127,7 +127,13 @@ export function ortbConverter({
           logError('ORTB response seatbid[].bid[].impid does not match any imp in request; ignoring bid', bid);
         })
       ).filter(Boolean);
-      return buildResponse(bidResponses, response, augmentContext(ctx.req));
+      const bidNonResposes = ((response.ext && response.ext.seatnonbid) || []).flatMap(seatnonbid =>
```

Comment:
I don't like this; nonbids are not bids, I don't think they should be mixed together. It also requires special handling from every adapter that happens to talk to a PBS instance (like Rubicon).

I think this should be a new "response processor", part of the PBS processors, like [this one](https://github.com/prebid/Prebid.js/blob/master/libraries/pbsExtensions/processors/pbs.js#L88). This way, every adapter that wants the PBS-specific dialect (including PBS and rubicon) would get this logic without any changes.

If you want/need to map these back to the original imp you'll need some changes in here that are also already in https://github.com/prebid/Prebid.js/pull/9342 - it might be easier to merge that in first.

Reply:
I agree with this. I think the new revision is much cleaner. I don't have a super clear picture of what exact data we want to be emitting that might be missing from the server response, so I didn't mess with response processing. But if there is a lot of data we'd want to tack on, I agree adding it in a response processor would be be good.